### ISSUE TITLE
Redis Connection Issue 

### ISSUE THREAD

**Issue by Alex2Tacy** [Sanjeev-Thiyagarajan/node-docker#1]
Hi , I'm following the tutorial from https://www.youtube.com/watch?v=9zUHg7xjIqQ&t=10521s . I encounter Redis connection issue below at 3:21:17 when I'm trying to do a POST.

ClientClosedError: The client is closed
    at Commander._RedisClient_sendCommand (/app/node_modules/@node-redis/client/dist/lib/client/index.js:408:31)
    at Commander.commandsExecutor (/app/node_modules/@node-redis/client/dist/lib/client/index.js:166:154)
    at Commander.BaseClass.<computed> [as set] (/app/node_modules/@node-redis/client/dist/lib/commander.js:8:29)
    at RedisStore.set (/app/node_modules/connect-redis/lib/connect-redis.js:65:21)
    at Session.save (/app/node_modules/express-session/session/session.js:72:25)
    at Session.save (/app/node_modules/express-session/index.js:406:15)
    at ServerResponse.end (/app/node_modules/express-session/index.js:335:21)
    at ServerResponse.send (/app/node_modules/express/lib/response.js:221:10)
    at ServerResponse.json (/app/node_modules/express/lib/response.js:267:15)
    at exports.login (/app/controllers/authController.js:45:28)
[nodemon] app crashed - waiting for file changes before starting...

I try to to add in the line  redisClient.connect().catch(console.error); but encounter another issue 

Error: connect ECONNREFUSED 127.0.0.1:6379
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1138:16) {
  errno: -111,
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '127.0.0.1',
  port: 6379
}

**Comment by brimarq** [comment]
I ran into the same issue. If you're using redis v4.x.x, you'll need to refactor `redisClient` to work with the newer version that no longer connects to the server automatically:

```
let redisClient = redis.createClient({
  legacyMode: true,
  socket: {
    host: REDIS_URL,
    port: REDIS_PORT,
  },
});
redisClient.connect().catch(console.error);
```
See the [connect-redis API for redis@v4](https://github.com/tj/connect-redis#api) and [node-redis migration guide](https://github.com/redis/node-redis/blob/master/docs/v3-to-v4.md).
